src/Mollie/WC/Helper/PaymentMethodIconUrl.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+use Mollie\Api\Types\PaymentMethod;
+
+class Mollie_WC_Helper_PaymentMethodIconUrl
+{
+    /**
+     * @var null
+     */
+    private static $instance = null;
+    /**
+     * @var array
+     */
+    private $paymentMethodImages;
+
+    /**
+     * PaymentMethodIconUrl constructor.
+     * @param array $paymentMethodImages
+     */
+    private function __construct(array $paymentMethodImages)
+    {
+        $this->paymentMethodImages = $paymentMethodImages;
+    }
+
+    /**
+     * @param $paymentMethodImages
+     * @return Mollie_WC_Helper_PaymentMethodIconUrl|null
+     */
+    public static function getInstance($paymentMethodImages)
+    {
+        if (self::$instance == null)
+        {
+            self::$instance = new Mollie_WC_Helper_PaymentMethodIconUrl($paymentMethodImages);
+        }
+
+        return self::$instance;
+    }
```

Comment:
I would prefer we do not introduce more singletons. 

Wouldn't be better if this class could be simply consumed in the following way?

```
$paymentMethodsList = Mollie_WC_Plugin::paymentMethodsImages();
$factory = new PaymentMethodIconUrl($paymentMethodsList);
```

Even though I think we should discuss about the differences between _Gateway_ and _Payment Methods_ and use them correctly. Currently seems them are interchangable.